src/test/util/random.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_TEST_UTIL_RANDOM_H
+#define BITCOIN_TEST_UTIL_RANDOM_H
+
+#include <consensus/amount.h>
+#include <random.h>
+#include <test/util/setup_common.h>
+#include <uint256.h>
+
+#include <cstdint>
+
+static inline uint32_t InsecureRand32()
+{
+    return g_insecure_rand_ctx.rand32();
+}
+
+static inline uint256 InsecureRand256()
+{
+    return g_insecure_rand_ctx.rand256();
+}
+
+static inline uint64_t InsecureRandBits(int bits)
+{
+    return g_insecure_rand_ctx.randbits(bits);
+}
+
+static inline uint64_t InsecureRandRange(uint64_t range)
+{
+    return g_insecure_rand_ctx.randrange(range);
+}
+
+static inline bool InsecureRandBool()
+{
+    return g_insecure_rand_ctx.randbool();
+}
+
+static inline CAmount SemiRandMoneyAmount()
```

Comment:
If this is now just returning an amount in the uniform `[0 .. MAX_MONEY)` range, maybe rename to `InsecureRandMoneyAmount()`? Also, bit of a nit, but maybe have it return `InsecureRandRange(MAX_MONEY + CAmount(1))`, since `MAX_MONEY` is a valid `CAmount`?